explorer/interpreter/type_checker.cpp

```diff
@@ -2223,6 +2223,20 @@ auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
           e->set_value_category(ValueCategory::Let);
           return Success();
         }
+        case IntrinsicExpression::Intrinsic::ArraySz: {
+          if (args.size() != 1) {
+            return CompilationError(e->source_loc())
+                   << "__intrinsic_array_sz takes 1 argument, received "
+                   << args.size();
+          }
+          /*CARBON_RETURN_IF_ERROR(ExpectExactType(
```

Comment:
Did you mean to comment this out? If so, why?

Reply:
type checking is corrected